src/Plugins/ComplexCore/src/ComplexCore/Filters/Algorithms/GenerateColorTable.cpp

```diff
@@ -0,0 +1,262 @@
+    break;
+  }
+  case DataType::int32: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<Int32Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<int32>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::int64: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<Int64Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<int64>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::uint8: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<UInt8Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<uint8>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::uint16: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<UInt16Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<uint16>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::uint32: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<UInt32Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<uint32>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::uint64: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<UInt64Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<uint64>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::float32: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<Float32Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<float32>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::float64: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<Float64Array>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<float64>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  case DataType::boolean: {
+    const auto& selectedDataArray = m_DataStructure.getDataRefAs<BoolArray>(m_InputValues->SelectedDataArrayPath);
+    GenerateColorArray<bool>(selectedDataArray, m_InputValues->SelectedPreset["RGBPoints"], m_InputValues->RgbArrayPath, m_DataStructure);
+    break;
+  }
+  default:
+    return MakeErrorResult(-10000, fmt::format("The selected array '{}' does not have a compatible data type.", m_InputValues->SelectedDataArrayPath.toString()));
+  }
```

Comment:
@joeykleingers I feel like we should be able to use the ExecuteDataFunction in FilterUtilities.hpp here by just sending in the m_InputValues->SelectedDataArrayPath to the GenerateColorArray function and retrieving the proper array there.  That would simplify this whole section down to one line.